include/eve/detail/function/simd/x86/slide_left.hpp

```diff
@@ -0,0 +1,133 @@
+#include <eve/conditional.hpp>
+
+namespace eve::detail
+{
+  template<real_scalar_value T, typename N, x86_abi ABI, std::ptrdiff_t Shift>
+  EVE_FORCEINLINE wide<T,N,ABI> slide_left_ ( EVE_SUPPORTS(sse2_)
+                                            , wide<T,N,ABI> v, index_t<Shift>
+                                            ) noexcept
+  requires(Shift <= N::value)
+  {
+          if constexpr(Shift == 0)        return v;
+    else  if constexpr(Shift == N::value) return wide<T,N,ABI>{0};
+    else
+    {
+      if constexpr( std::same_as<ABI,x86_128_>)
+      {
+        constexpr auto shift = Shift*sizeof(T);
+        using i_t = as_integer_t<wide<T,N,ABI>, unsigned>;
+
+        auto const b  = bit_cast(v, as_<i_t>());
+        auto result = bit_cast(i_t(_mm_bsrli_si128( b, shift)), as(v));
+
+        // Mask noises from smaller sized registers
+        if constexpr(N::value < expected_cardinal_v<T,x86_128_>)
+        {
+          result &= keep_first(N::value-Shift).mask(as(result)).bits();
+        }
+
+        return result;
+      }
+      else if constexpr( std::same_as<ABI,x86_256_>)
+      {
+        if constexpr( current_api >= avx2)
+        {
+          constexpr auto offset = Shift * sizeof(T);
+          __m256i bv = _mm256_permute2x128_si256(v, v, 0x83);
+                if constexpr(offset == 16)  return bv;
+          else  if constexpr(offset <  16)  return _mm256_alignr_epi8(bv, v, offset);
+          else                              return _mm256_bsrli_epi128(bv, offset - 16);
+        }
+        else
+        {
+          constexpr auto shifted_bytes = sizeof(T)* Shift;
+          using f_t = typename wide<T,N,ABI>::template rebind<float>;
+          auto const w  = bit_cast(v, as_<f_t>{});
+          auto const s0 = _mm256_permute2f128_ps(w,w,0x81 );
+
+          if constexpr(shifted_bytes == 28)
+          {
+            return bit_cast(f_t(_mm256_permute_ps(_mm256_blend_ps(f_t(0), s0, 0x8), 0x93)), as(v));
+          }
+          else if constexpr(shifted_bytes == 24)
+          {
+            return bit_cast(f_t( _mm256_shuffle_ps(s0, f_t(0), 0xe)), as(v));
+          }
+          else  if constexpr(shifted_bytes == 20)
+          {
+            return bit_cast(f_t(_mm256_permute_ps(_mm256_blend_ps(f_t(0), s0, 0xe), 0x39)),as(v));
+          }
+          else  if constexpr(shifted_bytes == 16)
+          {
+            return bit_cast(f_t(s0), as(v));
+          }
+          else  if constexpr(shifted_bytes == 12)
+          {
+            return bit_cast(f_t(_mm256_permute_ps(_mm256_blend_ps(w, s0, 0x77), 0x93)), as(v));
+          }
+          else  if constexpr(shifted_bytes == 8 )
+          {
+            return bit_cast(f_t(_mm256_shuffle_ps(w, s0, 0x4e)), as(v));
+          }
+          else  if constexpr(shifted_bytes == 4 )
+          {
+            return bit_cast(f_t(_mm256_permute_ps(_mm256_blend_ps(w, s0, 0x11), 0x39)), as(v));
+          }
+          else
```

Comment:
I'm pretty sure that you can do 2 instructions and not 3 for anything divisible by 4 as I mentioned previously.